src/lib/tecs/TECS.cpp

```diff
@@ -778,13 +782,27 @@ void TECS::_setFastDescend(const float alt_setpoint, const float alt)
 {
 	if (_control_flag.airspeed_enabled && (_fast_descend_alt_err > FLT_EPSILON)
 	    && ((alt_setpoint + _fast_descend_alt_err) < alt)) {
-		_fast_descend = 1.f;
+		auto now = hrt_absolute_time();
+
+		if (_enabled_fast_descend_timestamp == 0U) {
+			_enabled_fast_descend_timestamp = now;
+		}
+
+		if ((now - _enabled_fast_descend_timestamp) < FAST_DESCEND_RAMP_UP_TIME) {
+			_fast_descend = max(_fast_descend, static_cast<float>(now - _enabled_fast_descend_timestamp) / static_cast<float>
```

Comment:
```suggestion
			_fast_descend = constrain(_fast_descend + static_cast<float>(now - _enabled_fast_descend_timestamp) / static_cast<float>(FAST_DESCEND_RAMP_UP_TIME), 0.f, 1.f);
```

That way it ramps from wherever the `_fast_descend` is. If `_fast_descend` starts at 0, then it takes `FAST_DESCEND_RAMP_UP_TIME` and if it's already at 0.5 it only takes half the time to finish the ramp instead of doing nothing during the first half.

Reply:
That way it ramps up more than linearly, as _fast_descend is already increased as well. Also the fast descend value can only be close to 1 or zero, it should not happen that it is any number in between, as on ramp down it is ramped based on the error.